### PR TITLE
Next hypre outerloop cuda cori umpire raja test

### PR DESCRIPTION
**Pull request by YiningQin**
Update Umpire memory management code in the HW3D model for HW3d GPU code running on cori-NERSC. It will also be used to elmpb model or 6-field on cori-NERSC.

### CODE REVIEW

**Comment by dschwoerer** [comment]
Should this be merged into next? Or is this ment to go into some hypre branch?

master is very like not applicable.

**Comment by bendudson** [comment]
I've changed the base to next-hypre-outerloop-cuda . This doesn't seem to contain any changes to the Umpire memory management; there are just some setup and build improvements from @jonesholger .

File: CMakeLists.txt [hunk 1]
```diff
@@ -562,6 +559,10 @@ function(bout_add_integrated_test TESTNAME)
   target_include_directories(${TESTNAME} PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
 
   if(BOUT_USE_CUDA)
+     # more apropo for CUDA 11 and updated cmake
```
Comment (bendudson) [inline CMakeLists.txt:562]: Probably best remove this commented-out code

File: cmake/FindPETSc.cmake [hunk 2]
```diff
@@ -121,6 +121,9 @@ if (PETSC_DIR AND NOT PETSC_ARCH)
   set (petscconf "NOTFOUND" CACHE INTERNAL "Scratch variable" FORCE)
 endif (PETSC_DIR AND NOT PETSC_ARCH)
 
+
+#message (STATUS "PETSC_ARCH ${PETSC_ARCH}")
```
Comment (bendudson) [inline cmake/FindPETSc.cmake:125]: Remove commented-out code, or uncomment if it's useful

**Comment by jonesholger** [comment]
It looks like this PR is a work-in-progress, as I don't see anything related to hw3d in the list of changes. Also this PR should target next-hypre-outerloop-cuda-cori and in turn next-hypre-outerloop-cuda-cori should target next-hypre-outerloop-cuda. At some point next-hypre-outerloop-cuda should be reconciled with next. 

next-hypre-outerloop-cuda-cori has some anachronisms only applicable to Cori such as FindPetsc which should be made more general. And this branch may be made obsolete by Perlmutter,

**Pull request closed without merging**